thoth/storages/graph/postgres.py

```diff
@@ -6751,6 +6753,32 @@ def delete_solved(self, *, os_name: str, os_version: str, python_version: str) -
                 .delete()
             )
 
+    def purge_solver_documents(
+        self, os_name: Optional[str] = None, os_version: Optional[str] = None, python_version: Optional[str] = None
+    ) -> int:
+        """Store and purge to be deleted solver documents to Ceph"""
+        self.connect()
+        solver_store = SolverResultsStore()
+        solver_store.connect()
+
+        target_prefix = f"{solver_store.ceph.prefix.rsplit('/', maxsplit=1)[0]}/solver-purge-{datetime2datetime_str()}/"
+        target_store = CephStore(prefix=target_prefix)
+        target_store.connect()
+
+        solver_document_ids = self.get_solver_run_document_ids_all(
+            os_version=os_version, os_name=os_name, python_version=python_version
+        )
+
+        deleted_solver_documents_count = 0
+
+        for solver_document_id in solver_document_ids:
+            document = solver_store.retrieve_document(document_id=solver_document_id)
+            target_store.store_document(document, document_id=solver_document_id)
+            solver_store.delete(object_key=solver_document_id)
```

Comment:
It looks like `SolverResultsStore` has no `delete` method. It might be a good idea to provide one or call `solver_store.ceph.delete` instead:

```suggestion
            solver_store.ceph.delete(object_key=solver_document_id)
```